nodejs20.x - migrate sample from aws-sdk v2 to v3

sannya-singal opened the pull request:
<!-- Please refer to the contribution guidelines before raising a PR: https://github.com/localstack/localstack/blob/master/docs/CONTRIBUTING.md -->

<!-- Why am I raising this PR? Add context such as related issues, PRs, or documentation. -->
## Motivation
The current transcription sample is not compatible with the latest node version. 

<!-- What changes does this PR make? How does LocalStack behave differently now? -->
## Changes
This PR: 
1. Adds support to the sample to run against node20.x
2. Migrates aws-sdk v2 to v3
3. The project now uses ESM module systems 
4. Update the node version used in github workflows pipeline

steffyP reviewed (approved):
Great work, @sannya-singal 🙌 🚀 Nice to see the sample working with Nodejs 20 🎉 

I only have two questions:
1) when running the action, there are still some warnings, e.g. 
```
Please update the following actions to use Node.js 20: actions/checkout@v3, actions/setup-python@v4. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.
```

I think we should use `actions/checkout@v4` and `actions/setup-python@v5 `. 
Could you test this, and see if there are still warnings? 🙏 

2) Not related to this PR, just something I stumbled across: There seem to be some changes in your private repo (where this one was forked from), that is not in the localstack-sample yet. Could you [create a PR](https://github.com/localstack-samples/sample-serverless-transcribe/compare/main...sannya-singal%3Asample-transcribe-app%3Amain) if the change is relevant?

sannya-singal commented:
Thanks @steffyP for the review 🙌 🙇‍♀️ 

I have removed all the soon to be deprecated github actions in the repository and updated the ones giving warnings, also changed the serverless version to use `3.38.0` as there was a recent release yesterday which is causing some failures here: https://github.com/localstack-samples/sample-serverless-transcribe/actions/runs/9185966236. 

Regarding the forked repo diff, its already included in here: https://github.com/localstack-samples/sample-serverless-transcribe/blob/7f84c717e1346210f6735eb22f59002468104245/src/presign.js#L10
